### PR TITLE
Fix trimming of lint docs

### LINKED ISSUE #79748: rustc lint listing code snippets don't have any indentation
It used to (**on stable and beta**) look like this:

image

But on **nightly** it looks like:

image

The source code for the snippet is:

`````rust
    /// ### Example
    ///
    /// ```rust
    /// if let _ = 123 {
    ///     println!("always runs!");
    /// }
    /// ```
`````

This is true for other code snippets in the lint listing, though the rest of the book (and other books) don't have the issue.

### PR DESCRIPTION
**Pull request by camelid**
Fixes #79748.

It was removing all the indentation before.

r? @Mark-Simulacrum

### CODE REVIEW

File: src/tools/lint-docs/src/lib.rs [hunk 1]
```diff
@@ -141,12 +141,11 @@ impl<'a> LintExtractor<'a> {
             let (doc, name) = loop {
                 match lines.next() {
                     Some((lineno, line)) => {
-                        let line = line.trim();
-                        if let Some(text) = line.strip_prefix("/// ") {
-                            doc_lines.push(text.trim().to_string());
```
Comment (camelid) [inline src/tools/lint-docs/src/lib.rs:146]: I'm not sure why this was `trim`ing the docs, but this is what was causing the lack of indentation.

File: src/tools/lint-docs/src/lib.rs [hunk 2]
```diff
@@ -141,12 +141,11 @@ impl<'a> LintExtractor<'a> {
...
-                        } else if line.starts_with("///") {
+                        if let Some(text) = line.trim_start().strip_prefix("/// ") {
+                            doc_lines.push(text.to_string());
+                        } else if line.trim_start().starts_with("///") {
                             doc_lines.push("".to_string());
```
Comment (bjorn3) [inline src/tools/lint-docs/src/lib.rs:147]: I don't think `///a` should push an empty line.

Comment (camelid) [reply src/tools/lint-docs/src/lib.rs:147]: Yeah, it does seem weird. @ehuss: did you do this intentionally or was it a mistake?

Comment (jyn514) [reply src/tools/lint-docs/src/lib.rs:147]: Maybe something like
```rust
if let Some(text) = line.trim_start().strip_prefix("///") {
    doc_lines.push(text.trim_start().to_string());
}

Comment (camelid) [reply src/tools/lint-docs/src/lib.rs:147]: No, I think that will cause the same issue that this is trying to fix.

**Comment by ehuss** [comment]
Thanks for taking a look at this. The second `trim` was a mistake in #79522.  I would kinda prefer something like the following change:

```diff
diff --git a/src/tools/lint-docs/src/lib.rs b/src/tools/lint-docs/src/lib.rs
index 326b7948098..ea54a351e03 100644
--- a/src/tools/lint-docs/src/lib.rs
+++ b/src/tools/lint-docs/src/lib.rs
@@ -143,8 +143,8 @@ fn lints_from_file(&self, path: &Path) -> Result<Vec<Lint>, Box<dyn Error>> {
                     Some((lineno, line)) => {
                         let line = line.trim();
                         if let Some(text) = line.strip_prefix("/// ") {
-                            doc_lines.push(text.trim().to_string());
-                        } else if line.starts_with("///") {
+                            doc_lines.push(text.to_string());
+                        } else if line == "///" {
                             doc_lines.push("".to_string());
                         } else if line.starts_with("// ") {
                             // Ignore comments.
```

**Comment by camelid** [comment]
@ehuss Could you explain why you prefer that change over this one?

**Comment by ehuss** [comment]
It's just an opinion that it keeps the code simpler.  There's no reason to differentiate between start/end trimming, so just the simple `trim` method is fine.  Also, `trim` only needs to be called once, not in every `if` clause.

**Comment by camelid** [comment]
Ah, I think I got mixed up and thought that what you suggested wouldn't work. I applied your patch. This should be ready to merge now!

**Pull request merged**